content/cif-connector/src/main/content/jcr_root/libs/commerce/gui/components/authoring/editor/pagepreview/clientlibs/pagepreview.js

```diff
@@ -24,58 +24,55 @@ window.CIF.PagePreview = {};
 
     var relPdpPreview = '.cq-commerce-pdp-preview-activator';
     var relPlpPreview = '.cq-commerce-plp-preview-activator';
+    var productPreviewServletUrl = '/bin/wcm/cif.previewproduct.html';
+    var categoryPreviewServletUrl = '/bin/wcm/cif.previewcategory.html';
 
-    var handlePreview = function(e, data) {
+    var handleProductPreview = function(e, data) {
+        handlePreview(data, productPreviewServletUrl);
+    };
+
+    var handleCategoryPreview = function(e, data) {
+        handlePreview(data, categoryPreviewServletUrl);
+    };
+
+    var handlePreview = function(data, previewServletUrl) {
         if (!data) {
             return;
         }
 
         var selections = data.selections;
-        var productSlug = null;
+        var identifier = null;
         if (Array.isArray(selections)) {
             if (selections.length > 0 && selections[0]) {
-                productSlug = selections[0].value;
+                identifier = selections[0].value;
             }
         } else if (selections) {
-            productSlug = selections.value;
+            identifier = selections.value;
         }
 
-        var url = Granite.author.ContentFrame.location;
-        var previewUrl = createPreviewUrl(url, productSlug);
-
-        if (previewUrl) {
-            window.open(previewUrl);
-        }
-    };
-
-    var createPreviewUrl = function(url, selector) {
-        if (!url || !selector) {
-            return null;
-        }
-
-        var indLastSlash = url.lastIndexOf('/');
-        if (indLastSlash < 0 || indLastSlash === url.length - 1) {
+        // Currently the picker returns only one identifier for the selected item.
+        // It can be one of the following: <id>, <url_key>, <path>, <sku>, <sku>#<variant_sku>
+        // This needs to be fixed in the future version so the picker will return multiple identifiers
+        var [slug, variantSku] = identifier ? identifier.split('#') : [];
```

Comment:
I think this is pretty much confusing, because this should be for `sku#variant_sku` but you use that `slug` variable name here.

Reply:
There is a broader issue here with the way the picker returns the selected items (hence the comment to fix it in the future). We cannot distinguish from the identifier whether it contains `url_key`, `sku` or other item identifiers. The best we can do here is to choose a better variable names ..maybe `[itemIdentifier, itemVariant]`?